dot/sync/service.go

```diff
@@ -92,6 +92,7 @@ type Strategy interface {
 	Process(results []*SyncTaskResult) (done bool, repChanges []Change, blocks []peer.ID, err error)
 	ShowMetrics()
 	IsSynced() bool
+	Result() any
```

Comment:
why do we need this? I don't see it called anywhere.

Reply:
This will be used by the state sync in the future. I can remove it from this PR and include it in a new one if you want